addons/pos_loyalty/models/pos_session.py

```diff
@@ -113,3 +114,33 @@ def _get_pos_ui_product_product(self, params):
         self._process_pos_ui_product_product(products)
         result.extend(products)
         return result
+
+    def _pos_data_process(self, loaded_data):
+        super()._pos_data_process(loaded_data)
+        config = loaded_data['pos.config']
+        programs = loaded_data['loyalty.program']
```

Comment:
your runbot is red because of this. It is because the key 'loyalty.program' is only set if there are programs, so you probably just need to check that the key exists:

if self.config_id.all_program_ids:
            result += [
                'loyalty.program',
                'loyalty.rule',
                'loyalty.reward',
            ]

Reply:
yeap, let's do that, thanks!